### ISSUE TITLE
Implement alternative loss functions for AAD

### ISSUE THREAD

**Issue by matwey** [snad-space/coniferest#250]
Currently, Hinge-loss has obvious drawback: it separates the nominal and anomaly data, but now tries to make them as distant as possible. So, scores distribution concentrates at both sides of tau threshold.

I propose to use cross-entropy as an alternative. Additionally, tau hyper-parameter would disappear.

**Comment by matwey** [comment]
@emilleishida I think we need your comment here

**Comment by emilleishida** [comment]
Summary after live conversation:

- naively I would say this seems a good idea to displace the distribution to the edges of the parameter space
- past experience makes me thing that there might be edge cases where the entropy will blow-up, however, I think it is work to make some tests, since I cannot isolate such cases from the top of my mind

**Comment by matwey** [comment]
The other possibility is to implement weights for all nodes (instead of leafs) as it was suggested in original code-base. This allows to generalize knowledge on all scales (not only on scale of typical leaf region). Before, we need to explore the data we usually deal with.
